### ISSUE TITLE
replicating oti_tree_id behaviour

### ISSUE THREAD

**Issue by kcranston** [OpenTreeOfLife/otindex#19]
In the nexsons, trees are a subresource of studies, so we uniquely identify a tree using a combination of studyID + treeID. I have the ottreeindex tree table set up with a simple auto-incrementing id column, a tree_label (the treeID from the nexson), a Foreign Key to study_id, plus a Unique constraint on (study_id, tree_label). OTI created a tree property `oti_tree_id` with form 'studyID_treeID'.

Turns out that the webapp uses oti's `find_trees` with the `oti_tree_id` property instead of the phylesystem `/study/{STUDY_ID}/tree/{TREE_ID}` method for efficiency in cases where it doesn't want the whole tree, but just some metadata about the tree. 

I'd like to be consistent about how we use treeIDs throughout the UI and the APIs, but also don't want to force users to download large JSON files to only extract a few fields.

**Comment by kcranston** [comment]
I note that the phylesystem_api /study method has an optional /meta syntax (see [phylesystem-api docs](https://github.com/OpenTreeOfLife/phylesystem-api/blob/master/docs/README.md#fine-grained-access-via-get)) that returns only metadata about a study. So, one option would be to implement the same for tree, i.e. `study/{study_id}/tree/{tree_id}/meta` to return only metadata about a tree rather than the whole tree.

**Comment by mtholder** [comment]
I don't think that this will be hard to add.
Can you run https://github.com/mtholder/peyotl/blob/master/scripts/phylesystem/ot_phylesystem_list_meta_for_tree.py 
(which requires a local copy of phylesystem) passing it a study_id@tree_id arg.  If that script returns the info that you want, then it should be pretty easy to package that as a function in peyotl and make it accessible via the phylesystem-api.
If that script is omitting some data want, let me know.

**Comment by kcranston** [comment]
Awesome! Pinging @jimallman to make sure this would work for the curator / webapp. These are the **excluded** properties:

u'^ot:rootNodeId',
u'nodeById',
u'edgeBySourceId',
u'^ot:inGroupClade',
u'@xsi:type',
u'^ot:branchLengthTimeUnit',
u'^ot:branchLengthDescription',
u'^ot:tag',
u'^ot:branchLengthMode'

**Comment by mtholder** [comment]
actually. we probably should not exclude `^ot:tag` that is easy to change.

**Comment by jimallman** [comment]
**TL;DR:** I believe the exclusions above are fine, since the current web-app code only uses the `find_trees` method to add a tree to a collection.

Here's [the code that runs when we fetch the current oti results](https://github.com/OpenTreeOfLife/opentree/blob/6ed80328443dbd18d25774c55c12a3e830dd8738/curator/static/js/curation-helpers.js#L870-L947). More specifically, [this short block applies when we find a valid tree](https://github.com/OpenTreeOfLife/opentree/blob/6ed80328443dbd18d25774c55c12a3e830dd8738/curator/static/js/curation-helpers.js#L906-L927). To add an entry to the collection, we need these bits of information:
  1. the study's ID (already in hand)
  2. the **study's ot:studyPublicationReference**
  3. the tree's ID (already in hand)
  4. the **tree's @label**
  5. ~~the **tree's ot:comment** (overall description)~~

Items 4 (or 3, if that's empty) and 2 are combined to build the default display name for the tree in our collection, in the form "TREE_LABEL_OR_ID (STUDY_COMPACT_REF)". For example [this tree](https://tree.opentreeoflife.org/curator/study/edit/ot_533?tab=trees) would be assigned the default name "Fig. 2 BEAST (Dantas, 2016)".

Item 5 (tree description) could be used as the default "description" for this tree in the collection. However, I see that **we don't currently provide any way to add or edit this field!** So the default description is open for suggestions. Perhaps we'd like to build this from [other tree properties in Nexson](https://github.com/OpenTreeOfLife/phylesystem-api/wiki/NexSON#table-i-predicate-vocabulary)? For now, the simplest path is to ignore item 5 and **use an empty string as the default description**; hopefully the user will add reasons for inclusion or other useful text here.

_SOMEDAY-MAYBE..._ If the user wants to "pin" a particular SHA for this tree, we could fetch them all in a separate operation and let the user choose one. Using the current code, this could be done by fetching the entire Nexson study and keeping only the `version_history` property. Or for more efficiency, we might build a tiny service that returns just a list of ordered SHAs (and commit messages) for a given study.

**Comment by josephwb** [comment]
:thumbsup:
